tls.go

```diff
@@ -11,33 +11,34 @@ import (
 	"net"
 	"time"
 
-	"github.com/mholt/caddy/caddyfile"
+	"github.com/emersion/maddy/config"
 )
 
-func setTLS(d caddyfile.Dispenser, config **tls.Config) error {
-	for d.Next() {
-		args := d.RemainingArgs()
-		switch len(args) {
-		case 1:
-			switch args[0] {
-			case "off":
-				*config = nil
-				return nil
-			case "self_signed":
-				if err := makeSelfSignedCert(*config); err != nil {
-					return err
-				}
-			}
-		case 2:
-			if cert, err := tls.LoadX509KeyPair(args[0], args[1]); err != nil {
-				return err
-			} else {
-				(*config).Certificates = append((*config).Certificates, cert)
+func tlsDirective(m *config.Map, node *config.Node) (interface{}, error) {
+	switch len(node.Args) {
+	case 1:
+		switch node.Args[0] {
+		case "off":
+			return nil, nil
+		case "self_signed":
+			cfg := &tls.Config{}
+			if err := makeSelfSignedCert(cfg); err != nil {
+				return nil, err
 			}
+			return &cfg, nil
+		}
+	case 2:
+		cfg := tls.Config{}
+		if cert, err := tls.LoadX509KeyPair(node.Args[0], node.Args[1]); err != nil {
+			return nil, err
+		} else {
+			cfg.Certificates = append(cfg.Certificates, cert)
```

Comment:
`cfg` is lost here, never returned

Reply:
Oops. Never tested that branch since config refactoring.